src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +144,1336 @@ static DamageTable D_80BBF858 = {
+
+    this->collider.dim.pos.x = world_pos.x;
+    this->collider.dim.pos.y = world_pos.y;
+    this->collider.dim.pos.z = world_pos.z;
+    this->collider.dim.radius = radius;
+    this->collider.dim.height = height;
+
+    // It is not clear what this is for. May be a bug.
+    if ((s32)this != -0x190) {
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    }
+}
+
+void EnSGoro_UpdateEyes(EnSGoro* this) {
+
+    s16 eyeCtr;
+
+    if (this->actionFlags & EN_S_GORO_ACTIONFLAG_EYESOPEN) {
+        if (this->eyeTimer == 0) {
+            eyeCtr = 0;
+        } else {
+            this->eyeTimer--;
+            eyeCtr = this->eyeTimer;
+        }
+
+        if (eyeCtr == 0) {
+            this->eyeTexIndex++;
+            if (this->eyeTexIndex >= EYETEX_INDEX_MAX) {
+                this->eyeTimer = Rand_S16Offset(30, 30);
+                this->eyeTexIndex = EN_S_GORO_EYETEX_OPEN;
+            }
+        }
+    }
+}
+
+void EnSGoro_UpdateActorFocus(EnSGoro* this) {
+
+    f32 yDelta;
+
+    if (this->actionFlags & EN_S_GORO_ACTIONFLAG_ROLLEDUP) {
+        yDelta = this->actor.shape.yOffset;
+    } else {
+        yDelta = 58.0f;
+    }
+
+    this->actor.focus.pos.x = this->actor.world.pos.x;
+    this->actor.focus.pos.y = this->actor.world.pos.y + yDelta;
+    this->actor.focus.pos.z = this->actor.world.pos.z;
+    this->actor.focus.rot.x = this->actor.world.rot.x;
+    this->actor.focus.rot.y = this->actor.world.rot.y;
+    this->actor.focus.rot.z = this->actor.world.rot.z;
+}
+
+s32 EnSGoro_UpdateRotationToPlayer(EnSGoro* this, PlayState* play) {
+
+    Player* player = GET_PLAYER(play);
+    Vec3f playerPos;
+    Vec3f thisPos;
+    s16 target = (this->actor.yawTowardsPlayer - this->bodyRotY) - this->actor.shape.rot.y;
+
+    Math_SmoothStepToS(&this->headRotY, target, 4, 0x2AA8, 1);
+    this->headRotY = CLAMP(this->headRotY, -0x1FFE, 0x1FFE);
+
+    target = (this->actor.yawTowardsPlayer - this->headRotY) - this->actor.shape.rot.y;
+    Math_SmoothStepToS(&this->bodyRotY, target, 4, 0x2AA8, 1);
+    this->bodyRotY = CLAMP(this->bodyRotY, -0x1C70, 0x1C70);
+
+    playerPos = player->actor.world.pos;
+    playerPos.y = player->bodyPartsPos[7].y + 3.0f;
+    thisPos = this->actor.world.pos;
+    thisPos.y += 70.0f;
+
+    target = Math_Vec3f_Pitch(&thisPos, &playerPos) - this->bodyRotZ;
+    Math_SmoothStepToS(&this->headRotZ, target, 4, 0x2AA8, 1);
+    this->headRotZ = CLAMP(this->headRotZ, -0x1C70, 0x1C70);
+
+    target = Math_Vec3f_Pitch(&thisPos, &playerPos) - this->headRotZ;
+    Math_SmoothStepToS(&this->bodyRotZ, target, 4, 0x2AA8, 1);
+    this->bodyRotZ = CLAMP(this->bodyRotZ, -0x1C70, 0x1C70);
+
+    return true;
+}
+
+s32 EnSGoro_UpdateAttentionTarget(EnSGoro* this, PlayState* play) {
+
+    s16 delayCounter;
+
+    if (this->loseAttentionTimer == 0) {
+        delayCounter = 0;
+    } else {
+        delayCounter = --this->loseAttentionTimer;
+    }
+
+    if (delayCounter != 0) {
```

Comment:
```suggestion
    if (DECR(this->loseAttentionTimer) != 0) {
```
Also looks like `DECR`.